lib/routes/mercari/util.ts

```diff
@@ -0,0 +1,311 @@
+        throw new Error('Expected INTEGER for R');
+    }
+    const rLen = readDerLength(der, offset);
+    offset += rLen.bytesRead;
+    const r = der.subarray(offset, offset + rLen.length);
+    offset += rLen.length;
+
+    if (der[offset++] !== 0x02) {
+        throw new Error('Expected INTEGER for S');
+    }
+    const sLen = readDerLength(der, offset);
+    offset += sLen.bytesRead;
+    const s = der.subarray(offset, offset + sLen.length);
+    offset += sLen.length;
+
+    if (offset !== der.length) {
+        throw new Error('Extra bytes in DER signature');
+    }
+
+    return {
+        r: fixBufferLength(r, 32),
+        s: fixBufferLength(s, 32),
+    };
+}
+
+function readDerLength(buf: Buffer, offset: number): { length: number; bytesRead: number } {
+    const byte = buf[offset];
+    if (byte < 0x80) {
+        return { length: byte, bytesRead: 1 };
+    }
+    const bytesCount = byte & 0x7F;
+    if (bytesCount > 4) {
+        throw new Error('DER length too long');
+    }
+
+    let length = 0;
+    for (let i = 0; i < bytesCount; i++) {
+        length = (length << 8) | buf[offset + 1 + i];
+    }
+    return { length, bytesRead: 1 + bytesCount };
+}
+
+function fixBufferLength(buffer: Buffer, length: number): Buffer {
+    if (buffer.length > length) {
+        const start = buffer.length - length;
+        return buffer.subarray(start);
+    }
+    if (buffer.length < length) {
+        return Buffer.concat([Uint8Array.from(Buffer.alloc(length - buffer.length)), Uint8Array.from(buffer)]);
+    }
+    return buffer;
+}
+
+function generateDPOP({ uuid, method, url }: { uuid: string; method: string; url: string }): string {
+    // Generate ECDSA key pair
+    const { privateKey, publicKey } = crypto.generateKeyPairSync('ec', {
+        namedCurve: 'prime256v1',
+    });
+
+    // Create JWT payload
+    const payload = {
+        iat: Math.floor(Date.now() / 1000),
+        jti: uuid,
+        htu: url,
+        htm: method.toUpperCase(),
+    };
+
+    // Create JWT header
+    const header = publicKeyToHeader(publicKey);
+
+    // Prepare signing input
+    const headerB64 = strToBase64URL(JSON.stringify(header));
+    const payloadB64 = strToBase64URL(JSON.stringify(payload));
+    const signingInput = `${headerB64}.${payloadB64}`;
+
+    // Sign the input
+    const sign = crypto.createSign('SHA256');
+    sign.update(signingInput);
+    const derSignature = sign.sign(privateKey);
+
+    // Process signature
+    const { r, s } = derDecode(derSignature);
+    const signature = bytesToBase64URL(Buffer.concat([Uint8Array.from(r), Uint8Array.from(s)]));
+
+    return `${signingInput}.${signature}`;
+}
+
+const fetchFromMercari = async <T>(url: string, data: any, method: 'POST' | 'GET' = 'POST'): Promise<T> => {
+    const DPOP = generateDPOP({
+        uuid: 'uuidv4()',
+        method,
+        url,
+    });
+
+    const headers = new Headers({
+        DPOP,
+        'X-Platform': 'web',
+        Accept: '*/*',
```

Comment:
Unnecessary header `accept`, it will be added by RSSHub [automatically](https://github.com/DIYgod/RSSHub/blob/fd065934a6a038fda55c449ddf564c012076ae3d/lib/utils/request-rewriter/get.ts#L50).